lib/project_types/script/layers/application/create_script.rb

```diff
@@ -47,22 +47,23 @@ def call(ctx:, language:, sparse_checkout_branch:, script_name:, extension_point
           private
 
           def install_dependencies(ctx, language, script_name, project_creator)
-            task_runner = Infrastructure::Languages::TaskRunner.for(ctx, language, script_name)
-            CLI::UI::Frame.open(ctx.message(
-              "core.git.pulling_from_to",
-              project_creator.sparse_checkout_repo,
-              script_name,
-            )) do
-              UI::StrictSpinner.spin(ctx.message(
-                "core.git.pulling",
-                project_creator.sparse_checkout_repo,
-                script_name,
-              )) do |spinner|
-                project_creator.setup_dependencies
-                spinner.update_title(ctx.message("core.git.pulled", script_name))
+            CLI::UI::Frame.open(ctx.message("script.create.creating")) do
+              CLI::UI::Frame.open(project_creator.create_start_message) do
+                UI::StrictSpinner.spin(project_creator.create_inprogress_message) do |spinner|
+                  project_creator.setup_dependencies
+                  spinner.update_title(project_creator.create_finished_message)
+                end
+              end
+
+              # This isn't ideal, but works for now.
+              # Ideally, CreateScript shouldn't need to know which ProjectCreators need dependencies installed.
+              # For that we'd want to have better composability,
+              # and omit the following steps when we have an "other" language script.
+              unless language == "other"
+                task_runner = Infrastructure::Languages::TaskRunner.for(ctx, language, script_name)
```

Comment:
This is a dumb question, but is there a reason that the `ProjectCreator` can't use a `TaskRunner` itself? It seems like it should have the details on how to successfully create a project, whether that entails fetching code or installing dependencies. If the project creator had a task runner, then we wouldn't need this conditional, since `CreateScript` could trust that it does the right thing (kinda like what was mentioned in POODR).

Reply:
I've thought about this a bit, and I know we chatted about it as well, but I think we should actually keep them separate.
I've written out my thoughts [in this doc](https://docs.google.com/document/d/1LdzIm8DNVImULSZ3HLmFeomnYKSY0ZAUul43300ukLg/edit?usp=sharing), but TLDR:
- ProjectCreator and TaskRunner do separate things
- We use Taskrunner elsewhere in the code (build and push, where the projectcreator doesn't exist) for these other things
- The DX suffers if I were to consider the PC+TR steps together as the "Creating script" step. It becomes less detailed as I try to hide more of the implementation details from our layer that handles the UI.

Reply:
It seems clear that we don't know the best way forward with designing this, maybe we need to just commit to a quick option that all of us may not agree with. Maybe afterwards someone could own rearchitecting the CLI if we still think it's a problem? We had some good ideas with events in our paring with Spencer which could be good to explore later. 

I still think the simplest solution that follows our existing pattern is to create a `WasmTaskRunner` or `NoopTaskRunner`. Theres a few cases where we need task runners, and I think this would be simplest for now and allow us to iterate faster. I'm envisioning a lot of similar problems when we try to resolve all the task runner usage like in `PushScript` and `BuildScript` 😬  This task runner would be pretty bare and it doesn't really fit the pattern where the task runner wraps an external tool, but there are already exceptions in existing task runners which I think it could be okay to ignore until we figure out the right abstraction. 

We could do something like this:

```ruby
class WasmTaskRunner
  def needs_dependencies_installed? # checked before running ProjectDependencies.install
    false
  end

  def needs_build? # checked before running BuildScript.call
    false
  end

  def library_version(_) # currently used during push
    "N/A"
  end

  def metadata
    # read the metadata.json file
  end

  # ...
end
```

Reply:
^ Quick note that I'm not suggesting that what I wrote there is the right long term solution, but its just one idea how we can keep things clean while minimizing changes in the meantime until we figure out the right solution. I'm just starting to think this is a bigger problem 🤔

Reply:
Yeah, it seems like we need to rethink how we approach our tasks and UI. I don't want to block this PR because of that so I agree that option 1 is the least bad until we can improve the architecture. Let's just make sure that we capture this in an issue so that we don't forget about improving it in the future 😄

Reply:
Not going to weigh in on how to address this in the short term. Probably best to pick whatever option seems to have the smallest blast radius but still kind of makes sense with the existing architecture.

However, I am interested in the longer term solution. If I have free time in the next few weeks I may try to prototype something myself and/or write something up about my thoughts on it. If someone else is interested in working on the that I'd be happy to, and would like to, help with the design of it, though.